wsd/apps/rest/v0/views/notification.py

```diff
@@ -21,3 +25,29 @@ class NotificationViewSet(BaseModelViewSet):
     def get_queryset(self):
         user = self.request.user
         return self.model.objects.filter(user=user) if user.is_authenticated else self.model.objects.none()
+
+    @extend_schema(
+        summary="Mark all notifications as read",
+        description="Marks all notifications for the current user as read",
+        responses={
+            200: {"type": "object", "properties": {"message": {"type": "string"}}},
+            401: None,
+            403: {"type": "object", "properties": {"detail": {"type": "string"}}},
+        },
+    )
+    @action(detail=False, methods=["post"])
+    def mark_all_as_read(self, request):
+        """Mark all notifications as read for the current user."""
+        user = request.user
+        if not user.is_authenticated:
+            return Response(
+                {"detail": "Authentication credentials were not provided."}, status=status.HTTP_401_UNAUTHORIZED
+            )
+
+        # Get all unread notifications for the user
```

Comment:
I think we can get rid of the comments for these as these are relatively easy to read